python/cuml/_common/base.pyx

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+
+import cuml._common.handle
+
+
+class Base:
+    """
+    Base class for all the ML algos. It handles some of the common operations
+    across all algos. Every ML algo class exposed at cython level must inherit
+    from this class.
+    """
+
+    def __init__(self, handle=None, random_state=None, verbose=False):
+        """
+        Constructor. All children must call init method of this base class!
+
+        Parameters
+        ----------
+        handle : cuml.handle.Handle
+               If it is None, a new one is created just for this class
+        random_state : int
```

Comment:
Does it make sense to expose random_state at this level? The type of the random state is probably algorithm dependent. Also finer grained control is needed. IIRC @vinaydes had some more input on this.

Reply:
Are we sure that any pseudo-random number generator that we may use will have single int as state? If we try to abstract state, it might become too complicated for no benefit. Therefore, it might be better to leave random_state out of here.

Reply:
you guys are probably right.

I was trying to hide some of the RNG's complexities in the base class itself (many algos will certainly need this flag). Now that I think of it, I'll remove this param from the base class. We can get to this when we notice any such repeated RNG stuff in the future.

Reply:
Reverted `random_state`. Resolving this conversation. Thanks for catching this, guys.